engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala

```diff
@@ -123,7 +123,33 @@ case class WorkflowExecutionActorData(workflowDescriptor: EngineWorkflowDescript
   }
 
   def mergeExecutionDiffs(diffs: Traversable[WorkflowExecutionDiff]): WorkflowExecutionActorData = {
-    diffs.foldLeft(this)((newData, diff) => newData.mergeExecutionDiff(diff))
+    preconsolidateDiffs(diffs) match {
+      case Some(preconsolidated) =>
+        this.mergeExecutionDiff(preconsolidated)
+      case None =>
+        diffs.foldLeft(this)((newData, diff) => newData.mergeExecutionDiff(diff))
+    }
+  }
+
+  def preconsolidateDiffs(diffs: Traversable[WorkflowExecutionDiff]): Option[WorkflowExecutionDiff] = {
+    val result = diffs.foldLeft(WorkflowExecutionDiff.empty)((acc, diff) => {
+      if (acc.executionStoreChanges.keySet.intersect(diff.executionStoreChanges.keySet).nonEmpty
+        || acc.jobKeyActorMappings.keySet.intersect(diff.jobKeyActorMappings.keySet).nonEmpty
+        || acc.valueStoreAdditions.keySet.intersect(diff.valueStoreAdditions.keySet).nonEmpty
+        || acc.cumulativeOutputsChanges.intersect(diff.cumulativeOutputsChanges).nonEmpty
+        || acc.rootAndSubworkflowIds.intersect(diff.rootAndSubworkflowIds).nonEmpty
+      ) return None
```

Comment:
tricky...

Reply:
Any action required/expected for trickiness?

Reply:
IMO the develop version should at least have a comment to point this out since it's a rarity in our code base. [Others](https://tpolecat.github.io/2014/05/09/return.html) have different and perhaps more strongly held opinions. 🙂 

> The return keyword is not “optional” or “inferred”; it changes the meaning of your program, and you should never use it.

Reply:
>Eliminating return from the Scala language would result in zero programs that could no longer be written.

Yeah but so would making it a punch-card turing machine...

Reply:
But yeah I actually agree, this is super ugly and I should refactor